test/review-model.js

```diff
@@ -1,132 +1,244 @@
+                        title: 'puff',
+                    },
+                })
+                const review = await Review.writeNew(
+                    henry.id,
+                    puff.id,
+                    null,
+                    'test'
+                )
+            } catch (error) {
+                expect(error.message).to.equal('a review requires a rating')
+            }
+        })
+    })
+
+    describe('Review Routes', () => {
+        beforeEach(async () => {
+            await db.sync({ force: true })
+            const user = await User.create({
+                email: 'rosie@snacker.com',
+                password: '123ert',
+            })
+            const product = await Product.create({
+                title: 'puff',
+                brand: 'stay-puft',
+                description: 'tasty',
+                price: 1.1,
+                country: 'usa',
+            })
+            await Review.writeNew(user.id, product.id, 4, 'So good!')
+        })
+        describe('GET', () => {
+            it('api/reviews', async () => {
+                const response = await app.get('/api/reviews')
+                const reviews = response.body
+                expect(response.status).to.equal(200)
+                expect(reviews).to.be.ok
+                expect(reviews).to.be.an('array')
+            })
+            it('api/reviews/:id', async () => {
+                const jack = await Review.create({
+                    email: 'jack@snacker.com',
+                    password: 'abc123',
+                })
+                const response = await app.get(`/api/reviews/${jack.id}`)
+                const review = response.body
+                expect(response.status).to.equal(200)
+                expect(review).to.be.ok
+                expect(review).to.be.an('object')
+            })
+        })
+        describe('POST', () => {
+            it('api/reviews', async () => {
+                const jeff = await User.create({
+                    email: 'jeff@snacker.com',
+                    password: 'jeff_pw',
+                })
+
+                const cheeto = await Product.create({
+                    title: 'cheeto',
+                    brand: 'stay-puft',
+                    description: 'tasty',
+                    price: 1.1,
+                    country: 'usa',
+                })
+
+                const response = await app.post('/api/reviews').send({
+                    userId: jeff.id,
+                    productId: cheeto.id,
+                    rating: 4,
+                    text: 'Really good snack',
+                })
+                const newReview = response.body
+                expect(response.status).to.equal(201)
+                expect(newReview).to.be.ok
+                expect(newReview).to.be.an('object')
+            })
+        })
+        describe('DELETE', () => {
+            it('api/reviews/:id', async () => {
+                const tempReview = await Review.create({
```

Comment:
Should this be user.create or review.create?

Reply:
Thank you, fixed that and also changed test to first check that review was initially created and then successfully deleted.